mysql-test/lib/mtr_report.pm

```diff
@@ -499,31 +499,49 @@ sub mtr_report_stats ($$$$) {
 
       my $test_result;
 
-      # if a test case has to be retried it should have the result MTR_RES_FAILED in jUnit XML
       if ($test->{'retries'} > 0) {
-        $test_result = "MTR_RES_FAILED";
+        $test_result = "MTR_RES_FAIL";
+        if ($test->{'failures'} > 1){
+          $test_result = "MTR_RES_RETRY_FAIL";
+        }
+        if ($test->{'result'} eq "MTR_RES_PASSED"){
+          $test_result = "MTR_RES_RETRY_PASS";
+        }
       } else {
         $test_result = $test->{'result'};
       }
+      if ($test_result eq "MTR_RES_PASSED" && defined($test->{'warnings'})){
+        $test_result = "MTR_RES_PASS_WARNINGS";
+      }
```

Comment:
This situation describes the case when warnings appear in the log after a successful test. These warnings will be added to the XML, and not just remain in the log

Reply:
Console log contains at least 4 statuses: [ fail ], [ retry-fail ] , [ pass ], [ retry-pass ] . I'm not completely sure that the new status "pass with warnings" is needed. Failure database viewer can detect this situation without a new status (if CDATA is present for [ pass ] test it means we need to mark test as "test with some issues, it should be checked". But new status allows to simplify request to test results DB ("select all tests with status != pass" instead of status != pass of CDATA is not empty). Another solution: new status can be introduced at the level of XML parser

Reply:
what mtr prints is for readability, for humans. for xml reports that are automatically parsed, processed, and indexed and should be searchable, this is just an unnecessary complication.

For example, when I'm searching for failed tests I don't want to search for, like, "status = MTR_RES_FAILED || status = MTR_RES_RETRY_FAILED"

Reply:
Removed all the MTR_RES_RETRY_* statuses.